pkg/client/clientset_generated/internalclientset/clientset.go

```diff
@@ -173,6 +175,9 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
 
 // NewForConfig creates a new Clientset for the given config.
 func NewForConfig(c *rest.Config) (*Clientset, error) {
+	if c == nil {
```

Comment:
I am not sure we should handle this case at all. If a nil is passed, this is a programming error and a panic is as good.

Reply:
I see your point if this were about passing `nil` directly. Here are my reasons for checking the errors:

- If we can predict a way to cause a panic by calling a public API, the external API should present explicit error return values instead of panicking, when possible.

- A popular way to use `NewForConfig` is by passing it a `config` parameter generated by `InClusterConfig`. If `InClusterConfig` can't load the in-cluster configuration for whatever reason and that error is not checked, `config` will be `nil`. I find this relatively likely to happen. 

- The error is much more clear than a panic and will likely save the programmer some time by being explicit about what's happening.

Reply:
No 3rdparty code should depend on a func to detect nil in the input IMO. Instead it should check that the in-cluster config is actually valid. Turning that into an error instead of a panic encourages to not check for valid data in 3rdparty code.

Reply:
Another argument: with "present explicit error return values instead of panicking" we will get an error return value in every function. We don't want that either. Errors should be for dynamic error situations, not for signalling wrong input. Wrong input should be could before.